pandas-stubs/core/series.pyi

```diff
@@ -239,7 +240,7 @@ class Series(IndexOpsMixin[S1], NDFrame):
     @overload
     def __new__(  # type: ignore[overload-overlap]
         cls,
-        data: _ListLike,
+        data: Iterable,
```

Comment:
Wouldn't mind widening `data` to `Any` when `dtype` is specified (there are combinations that will error but that is already the case)

Reply:
Wouldn't we be able to say the kinds of iterables that are allowed?  E.g., if the specified dtype is `TimestampDtypeArg`, then we know only certain kinds of iterables are allowed.

Reply:
I would prefer not doing that for now: there are so many coerciable types, that I assume it might create false negatives.

Reply:
OK, that makes sense.